### PR TITLE
Support for environment files in set_env

### LINKED ISSUE #1938: Support environment files and comments in setenv
As added in #1668.

### PR DESCRIPTION
**Pull request by gaborbernat**
Resolves #1938.

### CODE REVIEW

File: docs/config.rst [hunk 1]
```diff
@@ -270,7 +270,17 @@ Base options
...
 
-   A dictionary of environment variables to set when running commands in the tox environment.
+   A dictionary of environment variables to set when running commands in the tox environment. Lines starting with a
+   ``file|`` prefix are considered environment files to load with their location being the content minus this prefix.
+
+    .. note::
+
+       Environment files are processed using the following rules:
+
+       - blank lines ignored,
```
Comment (jugmac00) [inline docs/config.rst:280]:
```suggestion
       - blank lines are ignored,
```

File: docs/config.rst [hunk 2]
```diff
@@ -270,7 +270,17 @@ Base options
...
+       - lines starting with the ``#`` character are ignored,
+       - each line is in KEY=VALUE format; both the key and the value is stripped,
```
Comment (jugmac00) [inline docs/config.rst:282]:
```suggestion
       - each line is in KEY=VALUE format; both the key and the value are stripped,
```

File: docs/config.rst [hunk 3]
```diff
@@ -270,7 +270,17 @@ Base options
 .. conf::
    :keys: set_env, setenv
 
-   A dictionary of environment variables to set when running commands in the tox environment.
+   A dictionary of environment variables to set when running commands in the tox environment. Lines starting with a
+   ``file|`` prefix are considered environment files to load with their location being the content minus this prefix.
```
Comment (jugmac00) [inline docs/config.rst:274]:
```suggestion
   ``file|`` prefix define the location of environment files.
```

I think it is both clear that the file prefix is omitted and that a configuration file will be loaded If given.

What maybe is missing... When using a relative path, which base path is asumed? The directory of the tox configuration file?

And are placeholders like `toxinidir` supported?

**Review by jugmac00 — APPROVED** [review]
LGTM with some annotations

File: docs/config.rst (hunk 3 above)
Comment (gaborbernat) [reply docs/config.rst:274]: Within the file itself no, within the `file|` part yes.

Comment (gaborbernat) [reply docs/config.rst]: Actually, it wasn't, but now I've fixed that part.

**Pull request merged**